### DISCUSSION TITLE
How to undo accidental deletion of field - lost all categorization of issues

### DISCUSSION THREAD

**Discussion by ritchbui** [community/community#15198] [Projects and Issues]
Hi, I was setting up a new board view and added a field/column that I didn't wanted. I then "deleted" that field from the column dropdown menu, hoping that it would just remove the column from that board view. It instead deleted the field altogether, project wide, resulting in the loss of categorization of all my issues in the other views. What I meant to do was "hide field" which I didn't see at first.

We currently rely on that categorization a lot and I was wondering if there was a way to revert the board back and restore that field.  There are hundreds of issues, and relabelling them would be quite onerous and inaccurate. 

Thanks!

**Comment by chumashmescheems** [comment]
namespace GiveLove
{
    partial class Form2
    {
        /// <summary>
        /// Required designer variable.
        /// </summary>
        private System.ComponentModel.IContainer components = null;

        /// <summary>
        /// Clean up any resources being used.
        /// </summary>
        /// <param name="disposing">true if managed resources should be disposed; otherwise, false.</param>
        protected override void Dispose(bool disposing)
        {
            if (disposing && (components != null))
            {
                components.Dispose();
            }
            base.Dispose(disposing);
        }

        #region Windows Form Designer generated code

        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InitializeComponent()
        {
            this.components = new System.ComponentModel.Container();
            System.ComponentModel.ComponentResourceManager resources = new System.ComponentModel.ComponentResourceManager(typeof(Form2));
            this.label1 = new System.Windows.Forms.Label();
            this.button1 = new System.Windows.Forms.Button();
            this.notifyIcon1 = new System.Windows.Forms.NotifyIcon(this.components);
            this.SuspendLayout();
            // 
            // label1
            // 
            this.label1.AutoSize = true;
            this.label1.BackColor = System.Drawing.Color.Transparent;
            this.label1.Font = new System.Drawing.Font("Maiandra GD", 25.8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.label1.ForeColor = System.Drawing.Color.Red;
            this.label1.Location = new System.Drawing.Point(187, 112);
            this.label1.Name = "label1";
            this.label1.Size = new System.Drawing.Size(715, 51);
            this.label1.TabIndex = 0;
            this.label1.Text = "Tớ thích cậu! Cậu có thích tớ không?";
            // 
            // button1
            // 
            this.button1.AutoSize = true;
            this.button1.Font = new System.Drawing.Font("Maiandra GD", 16.2F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.button1.ForeColor = System.Drawing.Color.Red;
            this.button1.Location = new System.Drawing.Point(251, 217);
            this.button1.Name = "button1";
            this.button1.Size = new System.Drawing.Size(187, 44);
            this.button1.TabIndex = 1;
            this.button1.Text = "Tớ thích cậu";
            this.button1.UseVisualStyleBackColor = true;
            this.button1.Click += new System.EventHandler(this.button1_Click);
            // 
            // notifyIcon1
            // 
            this.notifyIcon1.Icon = ((System.Drawing.Icon)(resources.GetObject("notifyIcon1.Icon")));
            this.notifyIcon1.Text = "Give Love";
            this.notifyIcon1.Visible = true;
            // 
            // Form2
            // 
            this.AcceptButton = this.button1;
            this.AutoScaleDimensions = new System.Drawing.SizeF(8F, 16F);
            this.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font;
            this.BackgroundImage = global::GiveLove.Properties.Resources._764967;
            this.BackgroundImageLayout = System.Windows.Forms.ImageLayout.Stretch;
            this.ClientSize = new System.Drawing.Size(1102, 583);
            this.Controls.Add(this.button1);
            this.Controls.Add(this.label1);
            this.Icon = ((System.Drawing.Icon)(resources.GetObject("$this.Icon")));
            this.MaximizeBox = false;
            this.MaximumSize = new System.Drawing.Size(1120, 630);
            this.MinimumSize = new System.Drawing.Size(1120, 630);

**Comment by polamjag** [comment]
I too think this issue is VERY important.

The data in the fields does not keep a history of changes to the issue, like a label. If you delete the field itself, you will never know what information it contained.

If the field is like a category of task, you may still be able to re-determine it -- but for some fields, such as Sprint fields, that may not be possible.

I think that hiding a field and the allowance for accidental deletion is totally inadequate at the moment.

At the very least, there should be a warning in the dialog that deleting a field deletes all the data in that field in an unrecoverable manner. Also, if a field is accidentally deleted, it would be desirable to have a grace period in which the user can undo the deletion for a short period of time.

**Comment by tay1orjones** [comment]
This is a duplicate of https://github.com/community/community/discussions/13373
